wagtail/test/testapp/templates/tests/custom_button.html

```diff
@@ -0,0 +1,8 @@
+{% load wagtailadmin_tags %}
+<form action="{{ button.url }}" method="post">
+    <input type="hidden" name="user_pk" value="{{ user_pk }}">
+    <button type="submit" class="{{ button.classname }}" {{ button.base_attrs_string }}>
+        {% icon name=button.icon_name %}
+        {{ button.label }}
+    </button>
+</form>
```

Comment:
Per the example in #11915, minus the `csrf_token` since we haven't tackled the problem of passing the request object yet. That is the step 2 described in https://github.com/wagtail/wagtail/issues/11915#issuecomment-2093342399.